sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -1371,6 +1371,41 @@ public void testInsertWithNestedGeneratedColumn() {
         assertThat(printedTable(response.rows()), is("10| 11| 4\n"));
     }
 
+    @Test
+    public void testInsertWithNestedGeneratedColumnGivenValueMatches() {
```

Comment:
Do we need all these integration level tests? Looks like the unit tests already cover those cases?

Reply:
It's true that unit tests also cover these, 
but I would keep them for as long as we have two paths (insert into values, insert into select)
to be on the safe side.

What do you think ?

Reply:
They're an order of magnitude or more slower than unit tests, and we don't want the time it takes to run the full test suite to explode over time. 

So I think we should limit the integration tests to "Is the generated columns class integrated correctly" and test all the edge cases and scenarios within the unit tests.

> I would keep them for as long as we have two paths (insert into values, insert into select)
to be on the safe side.

The problem is that we hardly ever notice that tests have become obsolete.